module/epochs/machine_account_test.go

```diff
@@ -178,3 +182,33 @@ func TestMachineAccountChecking(t *testing.T) {
 		})
 	})
 }
+
+// TestBackoff tests the backoff config behaves as expected. In particular, once
+// we reach the cap duration, all future backoffs should be equal to the cap duration.
+func TestMachineAccountValidatorBackoff_Overflow(t *testing.T) {
+
+	// large initial value to force a quick overflow, if unhandled
+	backoff := retry.NewExponential(time.Hour * 1000)
+	backoff = retry.WithCappedDuration(time.Microsecond*10, backoff)
+	backoff = retry.WithJitterPercent(5, backoff) // 5% jitter
+
+	calls := 0
+	last := time.Now()
+	retry.Do(context.Background(), backoff, func(ctx context.Context) error {
+		calls++
+		fmt.Println(calls, time.Since(last))
+
+		// assert the time between calls is greater than the cap
+		// an overflow would
+		if calls > 1 {
+			assert.Greater(t, time.Since(last), time.Microsecond*9)
```

Comment:
Should we be checking if time.Since(last) is equal to 10 micro seconds after the first backoff ?

Reply:
Since there is jitter in the backoff after the duration cap, it won't be exactly 10µs

Reply:
Gotcha, could we check if time.Since(last) <= capped duration + max possible jitter ?

Reply:
Checking against an upper bound could cause a flakey test, since the test will sleep for at least `backoff`, but possibly more. I changed the test to check `backoff.Next` directly so we don't need to worry about inconsistencies of measuring wallclock time in the test.

Reply:
Looks good